gvr-eyepicking/app/src/main/java/org/gearvrf/gvreyepicking/ColorShader.java

```diff
@@ -19,36 +19,34 @@
 import org.gearvrf.GVRContext;
 import org.gearvrf.GVRMaterialMap;
 import org.gearvrf.GVRMaterialShaderManager;
-import org.gearvrf.GVRCustomMaterialShaderId;
+import org.gearvrf.GVRShaderTemplate;
+import org.gearvrf.R;
+import org.gearvrf.utility.TextFile;
 
-public class ColorShader {
+import android.content.Context;
 
-    public static final String COLOR_KEY = "color";
+import org.gearvrf.GVRCustomMaterialShaderId;
 
-    private static final String VERTEX_SHADER = "attribute vec4 a_position;\n"
-            + "uniform mat4 u_mvp;\n" //
-            + "void main() {\n" //
-            + "  gl_Position = u_mvp * a_position;\n" //
+public class ColorShader extends GVRShaderTemplate
+{
+    private static final String VERTEX_SHADER = "in vec4 a_position;\n"
+            + "uniform mat4 u_mvp;\n"
+            + "void main() {\n"
+            + "  gl_Position = u_mvp * a_position;\n"
             + "}\n";
 
     private static final String FRAGMENT_SHADER = "precision mediump float;\n"
-            + "uniform vec4 u_color;\n" //
-            + "void main() {\n" //
-            + "  gl_FragColor = u_color;\n" //
+            + "uniform vec4 u_color;\n"
+            + "out vec4 gl_FragColor;\n"
+            + "void main() {\n"
+            + "  gl_FragColor = u_color;\n"
```

Comment:
OK it is called fragColor now.

From: rahul [mailto:notifications@github.com]
Sent: Thursday, June 23, 2016 2:03 PM
To: gearvrf/GearVRf-Demos <GearVRf-Demos@noreply.github.com>
Cc: Nola Donato <nola.donato@samsung.com>; State change <state_change@noreply.github.com>
Subject: Re: [gearvrf/GearVRf-Demos] Pickevents (#204)


In gvr-eyepicking/app/src/main/java/org/gearvrf/gvreyepicking/ColorShader.java<https://github.com/gearvrf/GearVRf-Demos/pull/204#discussion_r68314628>:

>              + "}\n";

>

>      private static final String FRAGMENT_SHADER = "precision mediump float;\n"

> -            + "uniform vec4 u_color;\n" //

> -            + "void main() {\n" //

> -            + "  gl_FragColor = u_color;\n" //

> +            + "uniform vec4 u_color;\n"

> +            + "out vec4 gl_FragColor;\n"

> +            + "void main() {\n"

> +            + "  gl_FragColor = u_color;\n"

It was not the newline that fixed it, was the name change. Seems to be some issue with the name gl_FragColor. Could you change it to fragColor?

—
You are receiving this because you modified the open/close state.
Reply to this email directly, view it on GitHub<https://github.com/gearvrf/GearVRf-Demos/pull/204/files/28b5b36be6c51b0d5bf1dd382043e66b53794616#r68314628>, or mute the thread<https://github.com/notifications/unsubscribe/AQw1S1EtePbJKmXR5fVGI9UJ0rWexlWIks5qOvSXgaJpZM4I6RXp>.